### PR TITLE
feat: add count-chars

### PR DESCRIPTION
**Pull request by isabelroses**
:3

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -74,6 +74,14 @@ enum Commands {
 		#[clap(default_value = "-")]
 		text: MaybeStdin<String>,
 	},
+	// Count a given charchter
```
Comment (uncenter) [inline src/main.rs]:
```suggestion
	// Count a given character
```

File: src/main.rs [hunk 2]
```diff
@@ -74,6 +74,14 @@ enum Commands {
...
+	CountChar {
+		#[clap(default_value = "-")]
+		text: MaybeStdin<String>,
+
+		#[clap(default_value = " ")]
+		char: MaybeStdin<char>,
```
Comment (uncenter) [inline src/main.rs]: How would this be from stdin?

**Review by uncenter — CHANGES_REQUESTED** [review]

File: src/main.rs [hunk 3]
```diff
@@ -74,6 +74,14 @@ enum Commands {
 		#[clap(default_value = "-")]
 		text: MaybeStdin<String>,
 	},
+	// Count a given charchter
+	CountChar {
+		#[clap(default_value = "-")]
+		text: MaybeStdin<String>,
+
+		#[clap(default_value = " ")]
```
Comment (uncenter) [inline src/main.rs]: Maybe the default value is `""`, and it counts just the overall number of characters in the string?

Comment (isabelroses) [reply src/main.rs]: > error: invalid value '' for '[CHARACTER]': cannot parse char from empty string

File: src/main.rs (hunk 2 above)
Comment (isabelroses) [reply src/main.rs]: good point...

File: src/main.rs (hunk 3 above)
Comment (uncenter) [reply src/main.rs]: Yeah yeah I know but I was thinking we could add a check for `char == ""` and then just count the chars instead.

File: src/main.rs [hunk 4]
```diff
@@ -74,6 +74,14 @@ enum Commands {
 		#[clap(default_value = "-")]
 		text: MaybeStdin<String>,
 	},
+	// Count a given character
```
Comment (uncenter) [inline src/main.rs]:
```suggestion
	/// Count total characters or the occurrences of a given character
```

File: src/main.rs (hunk 3 above)
Comment (isabelroses) [reply src/main.rs]: > Yeah yeah I know but I was thinking we could add a check for `char == ""` and then just count the chars instead.

Used an optional instead.

File: src/main.rs [hunk 5]
```diff
@@ -74,6 +74,12 @@ enum Commands {
 		#[clap(default_value = "-")]
 		text: MaybeStdin<String>,
 	},
+	/// Count total characters or the occurrences of a given character
+	CountChar {
```
Comment (uncenter) [inline src/main.rs]:
```suggestion
	CountChars {
```

🤔 (btw dont accept this suggestion on the GitHub UI since you have to update it below as well!)

**Review by uncenter — APPROVED** [review]

**Pull request merged**